### PR TITLE
Fix subnet allocation bug

### PR DESCRIPTION
**Pull request by tuommaki**
When searching for free space for given network, one must take into account the
network size. When free IP range starts from /26 network and caller asked for
free /24 network, the network IP for allocated range must be selected
correctly.

### CODE REVIEW

File: ipam_test.go [hunk 1]
```diff
@@ -828,7 +863,7 @@ func ExampleFree() {
 	// Output:
 	// 10.4.0.0/24
 	// 10.4.1.0/28
-	// 10.4.1.16/24
+	// 10.4.2.0/24
```
Comment (tuommaki) [inline ipam_test.go]: @JosephSalisbury I believe this was accidentally wrong and there's nothing that specifically expects something like this?

**Review by calvix — APPROVED** [review]
lgtm if this fix the problem

**Review by MarcelMue — APPROVED** [review]
Makes sense to me, we talked about this briefly.

File: ipam_test.go (hunk 1 above)
Comment (JosephSalisbury) [reply ipam_test.go]: Yeah, that looks right to me. We obviously can't start a /24 at 10.4.1.16 :D

**Comment by MarcelMue** [comment]
Merging here. We want to get a fix out. Please ping me if you see something wrong with this PR.

**Pull request merged**